underscore.js

```diff
@@ -1009,11 +1009,21 @@
 
   // Convert an object into a list of `[key, value]` pairs.
   _.pairs = function(obj) {
-    var keys = _.keys(obj);
-    var length = keys.length;
-    var pairs = Array(length);
-    for (var i = 0; i < length; i++) {
-      pairs[i] = [keys[i], obj[keys[i]]];
+    var pairs, i, length;
+    if (_.isMap(obj)) {
+      var entries = obj.entries();
```

Comment:
See [npm/toArray](https://github.com/lodash/lodash/blob/4.11.2-npm/toArray.js#L42-L56) for an example of using a helper function like [mapToArray](https://github.com/lodash/lodash/blob/4.11.2-npm/_mapToArray.js).